spec/unit_tests/moab/storage_object_validator_spec.rb

```diff
@@ -15,104 +15,137 @@
     let(:verification_array) { storage_obj_validator1.validation_errors }
 
     it 'returns correct data structure' do
-      expect(storage_obj_validator1.validation_errors).to be_kind_of Array
+      expect(verification_array).to be_kind_of Array
     end
 
     it 'has item in the array that is the expected structure' do
       expect(verification_array[0]).to be_kind_of Hash
     end
-    context 'under version directory' do
-      it 'has missing items' do
-        # v0001
-        expect(verification_array[0]).to eq(Moab::StorageObjectValidator::MISSING_DIR => "Missing directory: [\"data\", \"manifests\"] Version: v0001")
+
+    context "invalid moab" do
+      context 'under version directory' do
+        it 'has missing items' do
+          # v0001
+          expect(verification_array[0]).to eq(Moab::StorageObjectValidator::MISSING_DIR => "Missing directory: [\"data\", \"manifests\"] Version: v0001")
+        end
+        it 'has unexpected directory' do
+          # v0002
+          expect(verification_array[1]).to eq(Moab::StorageObjectValidator::EXTRA_CHILD_DETECTED =>"Unexpected item in path: [\"extra_dir\"] Version: v0002")
+        end
+        it 'has unexpected file' do
+          # v0003
+          expect(verification_array[2]).to eq(Moab::StorageObjectValidator::EXTRA_CHILD_DETECTED =>"Unexpected item in path: [\"extra_file.txt\"] Version: v0003")
+        end
+        it 'has missing data directory' do
+          # v0004
+          expect(verification_array[3]).to eq(Moab::StorageObjectValidator::MISSING_DIR =>"Missing directory: [\"data\"] Version: v0004")
+        end
       end
-      it 'has unexpected directory' do
-        # v0002
-        expect(verification_array[1]).to eq(Moab::StorageObjectValidator::EXTRA_CHILD_DETECTED =>"Unexpected item in path: [\"extra_dir\"] Version: v0002")
+      context 'under data directory' do
+        it 'has unexpected file' do
+          # v0005
+          expect(verification_array[5]).to eq(Moab::StorageObjectValidator::EXTRA_CHILD_DETECTED =>"Unexpected item in path: [\"extra_file.txt\"] Version: v0005")
+        end
+        it 'has missing metadata files' do
+          # v0005
+          expect(verification_array[4]).to eq(Moab::StorageObjectValidator::NO_XML_FILES =>"Version: v0005 Missing all required metadata files")
+          # v0007
+          expect(verification_array[8]).to eq(Moab::StorageObjectValidator::NO_XML_FILES =>"Version: v0007 Missing all required metadata files")
+        end
+        it 'has missing content directory' do
+          # v0006
+          expect(verification_array[7]).to eq(Moab::StorageObjectValidator::MISSING_DIR =>"Missing directory: [\"content\", \"metadata\"] Version: v0006")
+        end
+        it 'has no required items' do
```

Comment:
here's the test:  does METADATA have required files?

Reply:
ah yea it shouldn't say required items it should say "only contains files"